File: dmarc_report_api/queries/shared_structures/category_totals.py
import graphene


class CategoryTotals(graphene.ObjectType):
    """
    This object displays the total amount of messages that fit into each category
    """

    pass_spf_only = graphene.Int(
        description="Amount of messages that are passing SPF, but failing DKIM."
    )
    pass_dkim_only = graphene.Int(
        description="Amount of messages that are passing DKIM, but failing SPF."
    )
    full_pass = graphene.Int(
        description="Amount of messages that have are passing SPF and DKIM."
    )
    fail = graphene.Int(description="Amount of messages that fail both SPF and DKIM.")

    def resolve_pass_spf_only(self: dict, info):
        return self.get("pass-spf-only", None)

    def resolve_pass_dkim_only(self: dict, info):
        return self.get("pass-dkim-only", None)

    def resolve_full_pass(self: dict, info):
        return self.get("pass", None)

    def resolve_fail(self: dict, info):
        return self.get("fail", None)

File: src/types/category-totals.js
const { GraphQLObjectType, GraphQLInt } = require('graphql')

const categoryTotalsType = new GraphQLObjectType({
  name: 'CategoryTotals',
  description:
    'This object displays the total amount of messages that fit into each category.',
  fields: () => ({
    passSpfOnly: {
      type: GraphQLInt,
      description: 'Amount of messages that are passing SPF, but failing DKIM.',
      resolve: async ({ passSpfOnly }) => passSpfOnly,
    },
    passDkimOnly: {
      type: GraphQLInt,
      description: 'Amount of messages that are passing DKIM, but failing SPF.',
      resolve: async ({ passDkimOnly }) => passDkimOnly,
    },
    fullPass: {
      type: GraphQLInt,
      description: 'Amount of messages that are passing SPF and DKIM.',
      resolve: async ({ pass }) => pass,
    },
    fail: {
      type: GraphQLInt,
      description: 'Amount of messages that fail both SPF and DKIM.',
      resolve: async ({ fail }) => fail,
    },
  }),
})

module.exports = {
  categoryTotalsType,
}
